aea/helpers/yaml_utils.py

```diff
@@ -0,0 +1,153 @@
+    envvar_key = "!envvar"
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the AEAYamlLoader.
+
+        It adds a YAML Loader constructor to use 'OderedDict' to load the files.
+        """
+        super().__init__(*args, **kwargs)
+        _AEAYamlLoader.add_constructor(
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, self._construct_mapping
+        )
+        _AEAYamlLoader.add_constructor(self.envvar_key, self._envvar_constructor)
+        self._add_implicit_resolver_if_not_present_already()
+
+    def _add_implicit_resolver_if_not_present_already(self) -> None:
+        """Add implicit resolver for environment variables, if not present already."""
+        if self.envvar_key in dict(self.yaml_implicit_resolvers.get(None, [])):
+            _AEAYamlLoader.add_implicit_resolver(
+                self.envvar_key, self.envvar_matcher, None
+            )
+
+    @staticmethod
+    def _construct_mapping(loader, node):
+        """Construct a YAML mapping with OrderedDict."""
+        object_pairs_hook = OrderedDict
+        loader.flatten_mapping(node)
+        return object_pairs_hook(loader.construct_pairs(node))
+
+    @staticmethod
+    def _envvar_constructor(_loader, node):  # pragma: no cover
+        """Extract the matched value, expand env variable, and replace the match."""
+        node_value = node.value
+        match = _AEAYamlLoader.envvar_matcher.match(node_value)
+        env_var = match.group()[2:-1]
+
+        # check for defaults
+        var_name, default_value = env_var.split(":")
+        var_name = var_name.strip()
+        default_value = default_value.strip()
+        var_value = os.getenv(var_name, default_value)
+        return var_value + node_value[match.end() :]
+
+
+class _AEAYamlDumper(yaml.SafeDumper):
+    """
+    Custom yaml.SafeDumper for the AEA framework.
+
+    It extends the default SafeDumper so to dump
+    YAML configurations while *following the order of the fields*.
+
+    This class is for internal usage only; please use
+    the public functions of the module 'yaml_dump' and 'yaml_dump_all'.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the AEAYamlDumper.
+
+        It adds a YAML Dumper representer to use 'OderedDict' to dump the files.
+        """
+        super().__init__(*args, **kwargs)
+        _AEAYamlDumper.add_representer(OrderedDict, self._dict_representer)
+
+    @staticmethod
+    def _dict_representer(dumper, data):
+        """Use a custom representer."""
+        return dumper.represent_mapping(
+            yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, data.items()
+        )
+
+
+def yaml_load(stream: TextIO) -> Dict[str, Any]:
+    """
+    Load a yaml from a file pointer in an ordered way.
+
+    :param stream: file pointer to the input file.
+    :return: the dictionary object with the YAML file content.
+    """
+    return yaml.load(stream, Loader=_AEAYamlLoader)  # nosec
```

Comment:
this is changed from `aea.helpers.base`, but you can't see the differences. Also each of the function below has changed analogously.